content/oci/oci.go

```diff
@@ -120,13 +129,46 @@ func (s *Store) Push(ctx context.Context, expected ocispec.Descriptor, reader io
 
 // Exists returns true if the described content exists.
 func (s *Store) Exists(ctx context.Context, target ocispec.Descriptor) (bool, error) {
+	s.sync.RLock()
+	defer s.sync.RUnlock()
+
 	return s.storage.Exists(ctx, target)
 }
 
+// Delete deletes the content matching the descriptor from the store. Delete may
+// fail on certain systems (i.e. NTFS), if there is a process (i.e. an unclosed
+// Reader) using `target`.
+func (s *Store) Delete(ctx context.Context, target ocispec.Descriptor) error {
+	s.sync.Lock()
+	defer s.sync.Unlock()
+
+	resolvers := s.tagResolver.Map()
+	untagged := false
+	for reference, desc := range resolvers {
+		if content.Equal(desc, target) {
+			s.tagResolver.Untag(reference)
+			untagged = true
+		}
+	}
+	if err := s.graph.Remove(ctx, target); err != nil {
+		return err
+	}
+	if untagged && s.AutoSaveIndex {
+		err := s.SaveIndex()
```

Comment:
`SaveIndex()` at line 320 also requires sync. But... beware of dead lock.

Reply:
added the lock.